Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -831,6 +812,32 @@ void GraphicsContextCG::fillRect(const FloatRect& rect)
     CGContextFillRect(context, rect);
 }
 
+void GraphicsContextCG::fillRect(const FloatRect& rect, Gradient& gradient, const AffineTransform& gradientSpaceTransform)
+{
+    CGContextRef context = platformContext();
```

Comment:
Can't just the caller do this:

```
CGContextStateSaver stateSaver(context);
context.setFillGradient(gradient, gradientSpaceTransform);
context.fillRect(rect);
```

Reply:
That is fully correct, however, I need it in a cross-platform fashion.

To achieve that before this patch, I simply adjusted the fill gradient to point to the stroke gradient, fill the rectangle, and swap back. To avoid that, I added the new fillRect variant.